spec/requests/providers/client_received_legal_helps_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'rails_helper'
+
+RSpec.describe 'provider own home requests', type: :request do
+  let(:legal_aid_application) { create :legal_aid_application }
+  let(:provider) { legal_aid_application.provider }
+
+  describe 'GET providers/client_received_legal_help' do
+    subject { get providers_legal_aid_application_client_received_legal_help_path(legal_aid_application) }
+
+    context 'when the provider is not authenticated' do
+      before { subject }
+      it_behaves_like 'a provider not authenticated'
+    end
+
+    context 'when the provider is authenticated' do
+      before do
+        login_as provider
+        subject
+      end
+
+      it 'returns http success' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      describe 'back link' do
+        it 'points to the check benefits page' do
+          expect(response.body).to have_back_link(providers_legal_aid_application_check_passported_answers_path(legal_aid_application))
+        end
+      end
+    end
+  end
+
+  describe 'PATCH providers/client_received_legal_help' do
+    subject { patch providers_legal_aid_application_client_received_legal_help_path(legal_aid_application), params: params.merge(submit_button) }
+    let(:client_received_legal_help) { false }
+    let(:application_purpose) { Faker::Lorem.paragraph }
+    let(:params) do
+      {
+        merits_assessment: {
+          client_received_legal_help: client_received_legal_help.to_s,
+          application_purpose: application_purpose
+        }
+      }
+    end
+
+    context 'when the provider is authenticated' do
+      before do
+        login_as provider
+        subject
+      end
+
+      context 'Continue button pressed' do
+        let(:submit_button) do
+          {
+            continue_button: 'Continue'
+          }
+        end
+
+        it 'updates the record' do
+          expect(legal_aid_application.merits_assessment.reload.client_received_legal_help).to eq(client_received_legal_help)
+          expect(legal_aid_application.merits_assessment.reload.application_purpose).to eq(application_purpose)
+        end
+
+        # TODO: remove this test and implement next one when next page is known
+        it 'returns next page placeholder' do
+          expect(response.body).to include('Placeholder: Are the proceedings, for which funding is being sought, currently, before the court?')
```

Comment:
I don't think there is a need to match the whole text. `include('Placeholder: Are the proceedings')` is enough, and means the specs won't fail if someone removes a comma.